src/averageAges.js

```diff
@@ -55,12 +81,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWithMothers = people
+    .filter(person => onlyWithSon
+      ? person.sex === MALE_SEX
+      : true)
+    .filter(person => people
+      .some(child => child.name === person.mother));
+
+  const AverageAgeDiff = peopleWithMothers
+    .reduce((year, child) => year + child.born - people
+      .find(mother => child.mother === mother.name).born, 0)
+  / peopleWithMothers.length;
```

Comment:
Create a separate variable for the ages array and use your new `calculateAverageAge` function

Reply:
could you explain in more detail, I don’t understand how I can get rid of the find in this case